toonz/sources/toonz/matchlinecommand.cpp

```diff
@@ -473,13 +473,15 @@ class MergeCmappedCommand final : public MenuItemHandler {
 
     cloneColumn(cells, newLevelPath);
 
+	QString newLevelPathString = QString::fromStdWString(newLevelPath.getWideString());
+
     it = indices.begin();
     ++it;
     for (int count = 0; it != indices.end();) {
       int index = *it;
       it++;
       mergeCmapped(destColumn, index - count,
-				   QString::fromStdWString(newLevelPath.getWideString()),
+				   it == indices.end() ? newLevelPathString : newLevelPathString + "_tmp",
```

Comment:
It seems that you are trying to set temporary name for intermediate levels.
However, as it now stands this line adds `_tmp` at the end of the file path (i.e. after the file extension). For example, if the `newLevelPath` is `+drawings/A_merged.tlv` , temporary path becomes `+drawings/A_merged.tlv_tmp` which seems to have no effect in the level name.

How about modifying this part like as follows? :

```
ndex 49bcd65..c0a2484 100644
@@ -473,8 +473,7 @@ public:
 
     cloneColumn(cells, newLevelPath);
 
-    QString newLevelPathString =
-        QString::fromStdWString(newLevelPath.getWideString());
+    TFilePath tmpPath = newLevelPath.withName(QString::fromStdString(newLevelPath.getName() + "_tmp").toStdWString());
 
     it = indices.begin();
     ++it;
@@ -482,8 +481,8 @@ public:
       int index = *it;
       it++;
       mergeCmapped(destColumn, index - count,
-                   it == indices.end() ? newLevelPathString
-                                       : newLevelPathString + "_tmp",
+                   it == indices.end() ? newLevelPath.getQString()
+                                       : tmpPath.getQString(),
                    false);
       ColumnCmd::deleteColumn(index - count);
       progress.setValue(++count);
```

Reply:
Although both somehow seemed to work, your change makes more sense. I've made and commited the change per your suggestion.  Thanks!